addons/binding/org.openhab.binding.opendaikin/src/main/java/org/openhab/binding/opendaikin/internal/api/ControlInfo.java

```diff
@@ -0,0 +1,210 @@
+ *
+ */
+public class ControlInfo {
+    private static Logger logger = LoggerFactory.getLogger(ControlInfo.class);
+
+    public enum Mode {
+        UNKNOWN(-1),
+        AUTO(0),
+        DEHUMIDIFIER(2),
+        COLD(3),
+        HEAT(4),
+        FAN(6);
+
+        private int value;
+
+        Mode(int value) {
+            this.value = value;
+        }
+
+        public int getValue() {
+            return value;
+        }
+
+        public static Mode fromValue(int value) {
+            for (Mode m : Mode.values()) {
+                if (m.getValue() == value) {
+                    return m;
+                }
+            }
+
+            logger.debug("Unexpected Mode value of \"{}\"", value);
+
+            // Default to auto
+            return AUTO;
+        }
+    }
+
+    public enum FanSpeed {
+        AUTO("A"),
+        SILENCE("B"),
+        LEVEL_1("3"),
+        LEVEL_2("4"),
+        LEVEL_3("5"),
+        LEVEL_4("6"),
+        LEVEL_5("7");
+
+        private String value;
+
+        FanSpeed(String value) {
+            this.value = value;
+        }
+
+        public String getValue() {
+            return value;
+        }
+
+        public static FanSpeed fromValue(String value) {
+            for (FanSpeed m : FanSpeed.values()) {
+                if (m.getValue().equals(value)) {
+                    return m;
+                }
+            }
+
+            logger.debug("Unexpected FanSpeed value of \"{}\"", value);
+
+            // Default to auto
+            return AUTO;
+        }
+    }
+
+    public enum FanMovement {
+        UNKNOWN(-1),
+        STOPPED(0),
+        VERTICAL(1),
+        HORIZONTAL(2),
+        VERTICAL_AND_HORIZONTAL(3);
+
+        private int value;
+
+        FanMovement(int value) {
+            this.value = value;
+        }
+
+        public int getValue() {
+            return value;
+        }
+
+        public static FanMovement fromValue(int value) {
+            for (FanMovement m : FanMovement.values()) {
+                if (m.getValue() == value) {
+                    return m;
+                }
+            }
+
+            logger.debug("Unexpected FanMovement value of \"{}\"", value);
+
+            // Default to stopped
+            return STOPPED;
+        }
+    }
+
+    public boolean power = false;
+    public Mode mode = Mode.AUTO;
+    /** Degrees in Celsius. */
+    public Optional<Double> temp = Optional.empty();
+    public FanSpeed fanSpeed = FanSpeed.AUTO;
+    public FanMovement fanMovement = FanMovement.STOPPED;
+    /* Not supported by all units. Sets the target humidity for dehumidifying. */
+    public Optional<Integer> targetHumidity = Optional.empty();
+
+    private ControlInfo() {
+    }
+
+    public static ControlInfo parse(String response) {
+        logger.debug("Parsing string: \"{}\"", response);
+        ControlInfo info = new ControlInfo();
+
+        for (String keyValuePair : response.split(",")) {
```

Comment:
This seems duplicates / copied, could you maybe just extract the logic?

If it would just convert it to a map you could implement it here to.

```
info.pow = "1".equals(values.get("pow"));
info.mode = Optional.ofNullable(values.get("mode"))
    .flatMap(ControlInfo::parseInt)
    .map(Mode::fromValue)
    .orElse(Mode.AUTO);
info.mode = Optional.ofNullable(values.get("mode"))
    .flatMap(ControlInfo::parseInt)
    .map(Mode::fromValue)
    .orElse(Mode.AUTO);
```
You could then also lose the defaults